src/App.tsx

```diff
@@ -1,156 +1,237 @@
+    } else {
+      todo.title = editedValue;
+      todo.isEditing = false;
+    }
+
+    setIsEditing(false);
+    setEditedValue('');
+  };
+
+  const visibleTodos: Todo[] = useMemo(() => {
+    if (filter !== 'All') {
+      if (filter === 'Active') {
+        return [...todos.filter(todo => !completedTodos.includes(todo))];
+      } else {
+        return [...completedTodos];
+      }
+    } else {
+      return [...todos];
+    }
+  }, [todos, filter, completedTodos, isEditing]);
+
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
         <header className="todoapp__header">
-          {/* this button should have `active` class only if all todos are completed */}
-          <button
-            type="button"
-            className="todoapp__toggle-all active"
-            data-cy="ToggleAllButton"
-          />
-
-          {/* Add a todo on form submit */}
-          <form>
+          {todos.length > 0 && (
+            <button
+              type="button"
+              className={classNames('todoapp__toggle-all', {
+                active: !isVisible,
+              })}
+              data-cy="ToggleAllButton"
+              onClick={() => setIsVisible(!isVisible)}
+            />
+          )}
+
+          <form onSubmit={formSubmitHandler}>
             <input
               data-cy="NewTodoField"
+              value={inputValue}
+              onChange={event => setInputValue(event.target.value)}
               type="text"
               className="todoapp__new-todo"
               placeholder="What needs to be done?"
             />
           </form>
         </header>
 
-        <section className="todoapp__main" data-cy="TodoList">
-          {/* This is a completed todo */}
-          <div data-cy="Todo" className="todo completed">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-                checked
-              />
-            </label>
-
-            <span data-cy="TodoTitle" className="todo__title">
-              Completed Todo
+        {isVisible && (
+          <section className="todoapp__main" data-cy="TodoList">
+            {visibleTodos.map(todo => (
+              <div
+                key={todo.id}
+                data-cy="Todo"
+                className={classNames('todo', {
+                  completed: completedTodos.includes(todo),
+                })}
+              >
+                {todo.isEditing ? (
+                  <form
+                    style={{ gridColumn: '1 / 3' }}
+                    onSubmit={event => editHandler(event, todo)}
+                  >
+                    <input
+                      onBlur={() => {
+                        todo.isEditing = false;
+                        setEditedValue('');
+                        setIsEditing(false);
```

Comment:
you should apply edited value as new title for edited todo and change it in the `todos` state array